test/BasicNetworkTest.template.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Network } from '@core';
+import { expect } from 'chai';
+
+export const AssertNetwork = (network: Network, expectedName: string, expectedIdentifier: number): void => {
+    it('correct predefined networks are registered', () => {
+        // Assert:
+        expect(network.name).to.be.equal(expectedName);
+        expect(network.identifier).to.be.equal(expectedIdentifier);
+    });
+};
+
+export const BasicNetworkTester = <T extends Network>(networks: readonly T[], name: string, identifier: number): void => {
+    it('can find well known network by name', () => {
+        // Act:
+        const network = Network.findByName(networks, name);
+
+        // Assert:
+        AssertNetwork(network!, name, identifier);
```

Comment:
why `network!` ?

Reply:
The `network` possibly returns `undefined` value.
Typescript raises the error message `Type 'undefined' is not assignable to type 'Network'.`

So added `non-null-assertion-operator` to prevent the error message.  I read the information from here.
https://learntypescript.dev/07/l2-non-null-assertion-operator#non-null-assertion-operator-syntax

Reply:
ok, that makes sense 👍